Tempura/UITests/UIViewControllerTestCase.swift

```diff
@@ -0,0 +1,171 @@
+  /// We cannot instantiate it as you can use your own UIViewController subclass.
+  var viewController: VC { get }
+  
+  /// configure the VC for the specified `testCase`
+  /// this is typically used to manually configure properties to all the children VCs or Views.
+  func configure(vc: VC, for testCase: String)
+}
+
+
+public extension UIViewControllerTestCase where Self: XCTestCase {
+  func uiTest(testCases: [String], context: UITests.VCContext<VC>) {
+    let screenSizeDescription: String = "\(UIScreen.main.bounds.size)"
+    let descriptions: [String: String] = Dictionary(uniqueKeysWithValues: testCases.map { identifier in
+      let description = "\(identifier) \(screenSizeDescription)"
+      return (identifier, description)
+    })
+    let expectations: [String: XCTestExpectation] = descriptions.mapValues { identifier in
+      return XCTestExpectation(description: description)
+    }
+
+    XCUIDevice.shared.orientation = context.orientation
+
+    DispatchQueue.global().async {
+      for identifier in testCases {
+        var contained: VC!
+        var container: UIViewController!
+        DispatchQueue.main.sync {
+          contained = self.viewController
+          container = context.container.container(for: contained)
+        }
+
+        guard let description = descriptions[identifier] else { continue }
+
+        let isViewReadyClosure: (UIView) -> Bool = { view in
+          var isOrientationCorrect = true
+
+          // read again in case some weird code changed it outside the ViewControllerTestCase APIs
+          let isViewInPortrait = view.frame.size.height > view.frame.size.width
+
+          if context.orientation.isPortrait {
+            isOrientationCorrect = isViewInPortrait
+          } else if context.orientation.isLandscape {
+            isOrientationCorrect = !isViewInPortrait
+          }
+
+          let isReady = isOrientationCorrect && self.typeErasedIsViewReady(view, identifier: identifier)
+
+          return isReady
+        }
+
+        let dispatchGroup = DispatchGroup()
+        dispatchGroup.enter()
+        DispatchQueue.main.async {
+          UITests.asyncSnapshot(
+            view: container.view,
+            viewToWaitFor: contained.view,
+            description: description,
+            configureClosure: {
+              self.typeErasedConfigure(contained, identifier: identifier)
+            },
+            isViewReadyClosure: isViewReadyClosure,
+            shouldRenderSafeArea: context.renderSafeArea,
+            keyboardVisibility: context.keyboardVisibility(identifier)
+            ) {
+            // ScrollViews snapshot
+            self.scrollViewsToTest(in: contained, identifier: identifier).forEach { entry in
+              UITests.snapshotScrollableContent(entry.value, description: "\(identifier)_scrollable_content \(screenSizeDescription)")
+            }
+            dispatchGroup.leave()
+            expectations[identifier]?.fulfill()
+          }
+        }
+
```

Comment:
I think it would be nice to try to not duplicate all the snapshot logic for both cases, if possible, because most of the code is not related to a particular kind of VC and probably can be reused.

Reply:
Done in `e0a81ec`